### ISSUE TITLE
Running custom Mask_RCNN inference in openCV 

### ISSUE THREAD

**Issue by EyGy** [matterport/Mask_RCNN#2836]
Hi everybody,
i know there are lot's of issues regarding this topic already (cause i read through them all in the last 2 painful days).
I managed to save my custom Mask_RCNN model in Keras (.h5) as seemingly working (.pb) and saved model format. However I failed when trying to load the model in openCV or when trying to convert it to onnx as a workaround.

Trying https://github.com/matterport/Mask_RCNN/issues/1569#issuecomment-569908346 led me to a .pb model file but the .pbtxt file seems to be corrupted and is not accepted as this error (also posted by @kaanaykutkabakci) appears:

`"cv2.error: OpenCV(4.1.2) C:\projects\opencv-python\opencv\modules\dnn\src\tensorflow\tf_io.cpp:54: error: (-2:Unspecified error) FAILED: ReadProtoFromTextFile(param_file, param). Failed to parse GraphDef file: ./IG/protobuf.pbtxt in function 'cv::dnn::ReadTFNetParamsFromTextFileOrDie'"`

Does anybody here has clue how to get a working .pbtxt ?
Or is there an actually working guide for getting my custom model running in openCV that i could follow?

Any help is very much appretiated, since I am running out of patience trying to solve this on my own :(

**Comment by ilknurakcay** [comment]
Hello @EyGy ,did you find any solution for problem?

**Comment by EyGy** [comment]
@ilknurakcay I am sorry but I have to dissapoint you - I wasted way too many hours for this, but did not manage to get it running in openCV.

**Comment by ilknurakcay** [comment]
@EyGy Thank u for the answer 
It seems impossible to run this model in Opencv, it wouldn't make sense for me to deal with this issue. In your mention under another issue, I saw that you need the model for instance segmentation. Do you have a suggestion regarding this?
I'm not sure if I should try running maskrcnn with opencv on a different model

**Comment by EyGy** [comment]
@ilknurakcay unfortunatly, i have no suggestion. After many hours of failing i got frustrated and put this project on halt. I plan on restarting it with a proper custom dataset and another framework like detectron 2 or something similiar. However, if you manage to get an instance segmentation model running in openCV I would really appreciate if you could post here or (dm me) some details of your setup.

**Comment by kaanaykutkabakci** [comment]
Guys, i have faced your mentioned problems 2 years ago. I could not find any solution in order to run keras model with opencv. I have built custom Mask-RCNN model with Tensorflow Object Detection API. You can try TF OD API.

**Comment by EyGy** [comment]
> Guys, i have faced your mentioned problems 2 years ago. I could not find any solution in order to run keras model with opencv. I have built custom Mask-RCNN model with Tensorflow Object Detection API. You can try TF OD API.

Thank you @karanchahal! Is generating both the .pb and the .pbtxt file easily possible in TF OD API ?
Then I guess i have to try using my own data and train again without keras.

**Comment by kaanaykutkabakci** [comment]
Yes, it is possible.

**Comment by xuanzhiliu** [comment]
> 

I also use OD API for mask-rcnn but the output result is not very well,do you have any methods to change it? thk
